src/main/kotlin/no/nav/syfo/consumer/DkifConsumer.kt

```diff
@@ -0,0 +1,105 @@
+package no.nav.syfo.consumer
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import io.ktor.client.HttpClient
+import io.ktor.client.call.receive
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.features.json.JacksonSerializer
+import io.ktor.client.features.json.JsonFeature
+import io.ktor.client.request.get
+import io.ktor.client.request.headers
+import io.ktor.client.statement.HttpResponse
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import no.nav.syfo.Environment
+import no.nav.syfo.auth.StsConsumer
+import no.nav.syfo.consumer.domain.DigitalKontaktinfo
+import no.nav.syfo.consumer.domain.DigitalKontaktinfoBolk
+import no.nav.syfo.consumer.pdl.APPLICATION_JSON
+import org.slf4j.LoggerFactory
+import java.lang.RuntimeException
+import java.util.UUID.randomUUID
+
+class DkifConsumer(env: Environment, stsConsumer: StsConsumer) {
+    private val client: HttpClient
+    private val stsConsumer: StsConsumer
+    private val dkifBasepath: String
+
+    init {
+        client = HttpClient(CIO) {
+            install(JsonFeature) {
+                serializer = JacksonSerializer {
+                    registerKotlinModule()
+                    registerModule(JavaTimeModule())
+                    configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                }
+            }
+        }
+        this.stsConsumer = stsConsumer
+        dkifBasepath = env.dkifUrl
+    }
+
+    suspend fun isBrukerReservert(aktorId: String) : DigitalKontaktinfo? {
+        val requestUrl = "$dkifBasepath/api/v1/personer/kontaktinformasjon"
+        val stsTokenString = "Bearer ${stsConsumer.getToken().access_token}"
+        val response: HttpResponse? = try {
+            client.get<HttpResponse>(requestUrl) {
+                headers {
+                    append(HttpHeaders.ContentType, APPLICATION_JSON)
+                    append(HttpHeaders.Authorization, stsTokenString)
+                    append(NAV_CONSUMER_ID_HEADER, ESYFOVARSEL_CONSUMER_ID)
+                    append(NAV_PERSONIDENTER_HEADER, aktorId)
+                    append(NAV_CALL_ID_HEADER, createCallId())
+                }
+            }
+        } catch (e: Exception) {
+            log.error("Error while calling DKIF: ${e.message}")
+            null
+        }
+
+        return when (response?.status) {
+            HttpStatusCode.OK -> {
+                val content = response.receive<DigitalKontaktinfoBolk>()
+                extractDataFromContent(content, aktorId)
+            }
+            HttpStatusCode.Unauthorized -> {
+                log.error("Could not get adressesperre from PDL: Unable to authorize")
```

Comment:
Skal det være `DKIF` i stedet for `PDL`?

Reply:
Ops..